syntax = "proto3";

package bannerpb;

option go_package = "pkg/proto/banner;bannerpb";

message Banner {
  string title = 1;
  string content = 2;
  string description = 3;
  string link = 4;
  string ownerID = 5;
  string max_price = 6;
  int64 id = 7;
}

message BannerWithMinPrice {
  string min_price = 1;
  int64 code = 2;
}

message BannerRequest {
  int64 id = 1;
}

message ActiveBanners {
  repeated int64 banner_id = 1;
}

service BannerService {
  rpc GetRandomBanner(BannerWithMinPrice) returns (Banner);
  rpc GetSuitableBanners(BannerWithMinPrice) returns (ActiveBanners);
  rpc GetBannerByID(BannerRequest) returns (Banner);
}